Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -735,9 +736,11 @@ public class PBXProjGenerator {
                 if dependency.copyPhase != nil {
                     // custom copy takes precedence
                     customCopyDependenciesReferences.append(embedFile)
-                } else if dependencyTarget.type.isExtension {
+                } else if dependencyTarget.type.isExtension && dependencyTarget.type != .extensionKitExtension {
                     // embed app extension
                     extensions.append(embedFile)
+                } else if dependencyTarget.type == .extensionKitExtension {
+                    extensionKitExtensions.append(embedFile)
```

Comment:
i think this part can be write like below since we may have other cases which `dependencyTarget.type.isExtension` is true but we need to see other properties.


```swift
else if dependencyTarget.type.isExtension {
     if dependencyTarget.type == .extensionKitExtension {
        extensionKitExtensions.append(embedFile)
     } else {
         extensions.append(embedFile)
     }
}
```